go.mod

```diff
@@ -2,11 +2,11 @@ module github.com/profitbricks/profitbricks-sdk-go/v5
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/go-resty/resty/v2 v2.0.0
+	github.com/go-resty/resty/v2 v2.3.0
 	github.com/jarcoal/httpmock v1.0.4
 	github.com/kr/pretty v0.1.0 // indirect
 	github.com/stretchr/testify v1.4.0
-	golang.org/x/net v0.0.0-20190827160401-ba9fcec4b297 // indirect
+	golang.org/x/net v0.0.0-20200602114024-627f9648deb9 // indirect
```

Comment:
nah, just remove all indirect deps, they won't come back

Reply:
Hey @piepmatz,

I am on golang 1.14.2 and I've just tried this... removed all indirect deps, ran go get -u all and they all came back.

Regards,

Alex

Reply:
Hi Alex,

From `go help modules`:

> Indirect requirements only arise when [...] explicitly upgrading a module's dependencies ahead of its own stated requirements.

That's what `go get -u all` does and that's why it's generally not recommended.

Let's use `golang.org/x/net` as example. It's an indirect dependency because our direct dependency `github.com/go-resty/resty/v2` needs it (as seen by `go mod why -m golang.org/x/net`). With no indirect dependencies in `go.mod` we'll use the `golang.org/x/net` version from timestamp `20200513185701` (as seen in `go.sum`). `go get -u all` says that we don't care about what `github.com/go-resty/resty/v2` wants, but that we want to use the latest version `20200602114024` (which might even be incompatible to `github.com/go-resty/resty/v2`!).

So it all boils down to the question whether we leave managing indirect dependencies to our direct dependencies or if we want to do that. In my opinion the latter is a bad choice: It quickly becomes lots of work and it might even break our direct dependencies.

So what's the right way? Remove all indirect dependencies from `go.mod`, update direct dependencies as required (e.g. `go get github.com/go-resty/resty/v2@latest`). Running `go mod tidy` won't bring the indirect dependencies back.

Reply:
Hi @piepmatz,

I see what you are saying, and I completely agree with you. Will definitely take advantage of your insights within the next release.

Thanks for clarifying!

Alex